src/component.js

```diff
@@ -18,12 +22,9 @@ function registerComponentMagicMethod() {
 
         const parentObserver = new MutationObserver(mutations => {
             for (let i = 0; i < mutations.length; i++) {
-                const closestParentComponent = mutations[i].target.closest('[x-data]')
-                if ((closestParentComponent && !closestParentComponent.isSameNode(parentComponent))) continue
-                if (!closestParentComponent.__x) {
-                    throw 'Error locating $parent data'
-                }
-                $el.$parent = allowTwoWayCommunication(closestParentComponent.__x.getUnobservedData(), parentComponent)
+                const mutatedComponent = mutations[i].target.closest('[x-data]')
+                if ((mutatedComponent && !mutatedComponent.isSameNode(parentComponent))) continue
+                $el.$parent = allowTwoWayCommunication(parentComponent.__x.getUnobservedData(), parentComponent)
```

Comment:
you can probably return earlier as soon as you find a valid mutation, we don't need to keep looping on the array

Reply:
Updated. Thanks